src/pages/staticPages/tools/oncoprinter/OncoprinterStore.ts

```diff
@@ -0,0 +1,311 @@
+import _ from "lodash";
+import {countMutations, mutationCountByPositionKey} from "../../../resultsView/mutationCountHelpers";
+import {Mutation, MutationCountByPosition} from "../../../../shared/api/generated/CBioPortalAPI";
+import {sendSentryMessage} from "../../../../shared/lib/tracking";
+import {recurrentHotspotFilter} from "../../../../shared/lib/AnnotationUtils";
+
+export type OncoprinterMutationAnnotationSettings = Pick<MutationAnnotationSettings, "ignoreUnknown" | "hotspots" | "cbioportalCount" | "cbioportalCountThreshold" | "oncoKb" | "driversAnnotated">;
+
+function genomeNexusKey(l:OncoprinterInputLineType3_Incomplete){
+    return `${l.chromosome}_${l.startPosition}_${l.endPosition}_${l.referenceAllele}_${l.variantAllele}`;
+}
+
+function genomeNexusKey2(l:{chromosome:string, start:number, end:number, referenceAllele:string, variantAllele:string}){
+    return `${l.chromosome}_${l.start}_${l.end}_${l.referenceAllele}_${l.variantAllele}`;
+}
+
+export default class OncoprinterStore {
+
+    // NOTE: we are not annotating hotspot because that needs nucleotide positions
+    //      we are not annotating COSMIC because that needs keywords
+
+    @observable.ref _sampleIdOrder:string | undefined = undefined;
+    @observable.ref _geneOrder:string | undefined = undefined;
+    @observable mutationAnnotationSettings:OncoprinterMutationAnnotationSettings = initMutationAnnotationSettings(this);
+    @observable.ref _dataInput:string|undefined = undefined;
+
+    @computed get didOncoKbFail() {
+        return this.oncoKbData.peekStatus === "complete" && (this.oncoKbData.result instanceof Error);
+    }
+
+    readonly sampleIds = remoteData({
+        await:()=>[this.parsedInputLines],
+        invoke:async()=>{
+            if (this.sampleIdOrder) {
+                return this.sampleIdOrder;
+            } else {
+                return getSampleIds(this.parsedInputLines.result!);
+            }
+        }
+    });
+
+    @action setSampleIdOrder(input:string) {
+        this._sampleIdOrder = input.trim();
+    }
+
+    @computed get sampleIdOrder() {
+        if (this._sampleIdOrder) {
+            return this._sampleIdOrder.split(/[,\s]+/);
+        } else {
+            return undefined;
+        }
+    }
+
+    @action setGeneOrder(input:string) {
+        this._geneOrder = input.trim();
+    }
+
+    @computed get geneOrder() {
+        if (this._geneOrder) {
+            return _.uniq(this._geneOrder.split(/[,\s]+/));
+        } else {
+            return undefined;
+        }
+    }
+
+    @action setDataInput(input:string) {
+        this._dataInput = input;
+    }
+
+    @action public setInput(data:string, genes:string, samples:string) {
+        this.setDataInput(data);
+        this.setGeneOrder(genes);
+        this.setSampleIdOrder(samples);
+    }
+
+    public readonly parsedInputLines = remoteData({
+        invoke:async()=>{
+            if (!this._dataInput) {
```

Comment:
seems like a this data shaping is worthy of testing and should be decoupled from promise wrapper and fetching.   but if it's gonna be replaced by the unified data interface you told me about then lets discuss what makes sense

Reply:
good point... probably worth doing now too, its not too much work